### PR TITLE
Corspman can now suture without fumbling

### PR DESCRIPTION
**Pull request by Ldip999**
## About The Pull Request

Corpsman can now suture without fumbling

## Why It's Good For The Game

Field surgery is already weak as is without a medical APC or medical tad (which is infrequently chosen as the tad usually is better off with protection instead), this change makes it so that corspman can suture without fumbling. If I remember correctly, their fumble time is 4 seconds as is, which is impractical because to extend a lifetime of a corpse you have to do CPR every 7 seconds and every precious second counts. 
While I don't believe most corpsman will begin carrying sutures in most cases as only having a corpse once or twice have over -100% health post kitting is rare on most OPs, and would instead probably be medevaced like usual, this is a buff to corpsman on maps where corpses regularly gain over 200 combined damage post kitting and the medevac is on cooldown for longer than the interval of corpses piling up (Think magmore engineering for an example, with marines being braziled into lava.) or crash/lowpop in general where we might not have shipside medical.
Note, that this is already how CM works, with corpsman being able to perform basic surgeries, such as suturing or internal bleeding, so for the balance side of things, I feel like they have already paved the path and playtested it for us, if that's any consideration towards approving and testmerging these changes.

## Changelog
:cl:
balance: Corpsman can now suture without fumbling
code: Dynamic surgery skill requirements
/:cl:

### CODE REVIEW

**Comment by Xander3359** [comment]
I disagree with this. Medics should not be carrying surgery tools because they are only trained for temporary field solutions. 

Also we are not CM and should not be compared to CM so that point is moot.

**Comment by jroinc** [comment]
How long does it take to suture, and how much does it heal? Would this just be a full, better, replacement for ATK's, or is the heal limited?

**Comment by jroinc** [comment]
looks like ~8 seconds and 15 heal for a corpsman? Seems reasonable, if those are the numbers.

**Comment by Ldip999** [comment]
> I disagree with this. Medics should not be carrying surgery tools because they are only trained for temporary field solutions.

Suturing IS a field solution

**Comment by jroinc** [comment]
Adds a surgery step that heals 10 external damage per use, performed with the new suture or, less optimally, a glass shard.
Replaces the drill in surgical vests and trays with the suture, since it's not really used for anything anyway. They're both printable in lathes.
Sprite by Fitz Sholl, thanks.
Why It's Good For The Game

Adds a treatment option for people who arrive in medbay with an absurd amount of damage that isn't just defibbing until they're alive.

> > I disagree with this. Medics should not be carrying surgery tools because they are only trained for temporary field solutions.
> 
> Suturing IS a field solution

Not as was originally added- that's a role-change. It's probably fine.

**Comment by Autisem** [comment]
suturing still causes fuck ton of infection so may wana reconsider doing that in the field without soap

**Comment by XElectricX** [comment]
Good change for reviving the horribly mangled bodies.

**Comment by yyzsong** [comment]
I disagree with this pr. Mangled corpses should be sent up to medbay. I don't think marines need a buff in this regard. Shipside medbay is already very slow and I can't bring myself to care about crash. Lowpop gamemodes will never be fun.

**Comment by Lumipharon** [comment]
I dislike it not because it's unbalanced, but it encourages medics to play as field doctors instead of medics.

Let doctors do their job

**Comment by XElectricX** [comment]
> play as field doctors
> Let doctors do their job

We're not even allowed to play that way thanks to the rules.

**Comment by Ldip999** [comment]
Medics will still fumble with all surgeries except suturing, and be much slower and worse than suturing than actual doctors. I feel like it would be griefing to do suturing when your medevac is off cooldown. As outlined previously, this is more for edge cases where the corpses pile on faster than your medevac.

**Review by Kuro020 — APPROVED** [review]

**Comment by Pariah919** [comment]
Should probably make suturing not cause infections, no idea why it even does that from a logical standpoint. Its not really compareable to normal surgery steps.

File: code/modules/surgery/surgery.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@ GLOBAL_LIST_INIT(surgery_steps, init_surgery())
 	var/list/allowed_species = null //List of names referencing species that this step applies to.
 	var/list/disallowed_species = null
 
-
+	var/surgery_skill_required = SKILL_SURGERY_PROFESSIONAL
```
Comment (Lumipharon) [inline code/modules/surgery/surgery.dm:28]: autodoc

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/surgery/surgery.dm (hunk 1 above)
Comment (Ldip999) [reply code/modules/surgery/surgery.dm:28]: What do you mean by autodoc?

Comment (Xander3359) [reply code/modules/surgery/surgery.dm:28]: You use /// to comment what the variable does on the line above the variable.
Auto-documentation

Comment (Ldip999) [reply code/modules/surgery/surgery.dm:28]: Oh! Automatice documentation! I thought you meant like the medbay machine autodoctor. I'm stupid.

File: code/modules/surgery/surgery.dm [hunk 2]
```diff
@@ -24,8 +24,8 @@ GLOBAL_LIST_INIT(surgery_steps, init_surgery())
 	var/list/allowed_tools = null //Array of type path referencing tools that can be used for this step, and how well are they suited for it
 	var/list/allowed_species = null //List of names referencing species that this step applies to.
 	var/list/disallowed_species = null
-
-
+	///Surgery skill required for this surgery step to start without the innitial fumble delay. By default it's SKILL_SURGERY_PROFESSIONAL (3). For reference, squaddies are 0, command is 1, corpsman is 2, researcher is 3, synths and doctors are 4, and CMO is 5
```
Comment (Lumipharon) [inline code/modules/surgery/surgery.dm:27]:
```suggestion
	///Surgery skill required for this surgery step to start without the innitial fumble delay
```

**Review by Lumipharon — CHANGES_REQUESTED** [review]
don't need giant autodoc, you're just explaining the var.

**Pull request merged**